0739-Daily-Temperatures.py

```diff
@@ -0,0 +1,18 @@
+'''
+Given a list of daily temperatures T, return a list such that, for each day in the input, tells you how many days you would have to wait until a warmer temperature. If there is no future day for which this is possible, put 0 instead.
+
+For example, given the list of temperatures T = [73, 74, 75, 71, 69, 72, 76, 73], your output should be [1, 1, 4, 2, 1, 1, 0, 0].
+
+Note: The length of temperatures will be in the range [1, 30000]. Each temperature will be an integer in the range [30, 100]. 
+'''
+class Solution:
+    def dailyTemperatures(self, T: List[int]) -> List[int]:
+        res = [0] * len(T)
+        stack = []
+        for i in range(len(T) - 1, -1, -1):
+            while stack and T[i] >= T[stack[-1]]:
+                stack.pop()
+            if stack:
+                res[i] = stack[-1] - i
+            stack.append(i)
+        return resa
```

Comment:
Time and Space complexity?

Reply:
Time Complexity: O(n)
Space Complexity: O(1) # Little confused about it though, might be wrong

Reply:
Why space complexity O(1)?

Reply:
Because we are using res for output
And then the size of stack will constant I guess, I mean, we are popping out everything
But then now I am confused as we have actually used some space
Will the space complexity be O(m) where m is the maximum length of stack.

Reply:
The size of the stack will not be constant it will depend upon the size of the array. Also in worst-case size will be O(n) so space complexity will be considered as O(n)